.github/workflows/build_and_test.yaml

```diff
@@ -1,10 +1,16 @@
 name: Build and Test
 on:
   push:
+    branches:
+      - "main"
+      - "release-v*"
```

Comment:
Personally I'd rather leave this running for all branches, at least for push builds, so CI runs on branches that I create in my fork, before I open a PR. Is there a specific reason for limiting to only these branches?

Reply:
limiting this so the CI runner cannot be misused
we should ensure local tooling such as `make check` & `make test` give devs enough confidence before raising a PR

Reply:
you can modify this file in your fork branch anyway?

Reply:
OK, not a big issue for me.